### PR TITLE
Adding json quasi quoter

### PR DESCRIPTION
**Pull request by nhibberd**
! @jystic @charleso

### CODE REVIEW

File: x-aeson/src/X/Data/Aeson/QQ.hs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import qualified Data.HashMap.Strict as HM
+import           Data.String (String)
+import qualified Data.Text as T
+import qualified Data.Vector as V
+
+import           Language.Haskell.TH (ExpQ)
+import           Language.Haskell.TH.Quote (QuasiQuoter (..), quoteDec, quotePat, quoteExp, quoteType)
+import           Language.Haskell.TH.Syntax (Q, Exp (..), Lit (..),  qAddDependentFile, runIO)
+
+import qualified Prelude (error)
```
Comment (amosr) [inline x-aeson/src/X/Data/Aeson/QQ.hs]: `as Savage`?

**Comment by jystic** [comment]
Some tests would be nice (underthebus)

As discussed IRL, maybe this would be better as a template haskell splice? I would expected a json quasi-quoter to take json rather than a file.

It shouldn't be too much different, [file-embed](http://hackage.haskell.org/package/file-embed-0.0.10/docs/src/Data-FileEmbed.html#embedFile) is an example of this kind of thing.

File: x-aeson/ambiata-x-aeson.cabal [hunk 2]
```diff
@@ -12,13 +12,16 @@ description:           x-aeson.
 
 library
   build-depends:
-                       aeson                           >= 0.8        && < 0.12
+                       ambiata-p
+                     , aeson                           >= 0.8        && < 0.12
```
Comment (tmcgilchrist) [inline x-aeson/ambiata-x-aeson.cabal]: `aeson` 1.0.1.0 is a thing, not sure if that's good or bad.

**Comment by charleso** [comment]
Agree with Jacob about QQ vs TH, was surprised when I saw it being used in bucket and assumed it would be a snippet of json and not a file

**Comment by nhibberd** [comment]
!! low

**Comment by jystic** [comment]
legit 👍

**Pull request merged**